ecllibrary/std/Type.ecl

```diff
@@ -0,0 +1,241 @@
+/*##############################################################################
+## HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.  All rights reserved.
+############################################################################## */
+
+
+EXPORT Type := MODULE
+
+  /**
+   * Type.vrec(x) deals with translations to arbitrary record type x
+   * MORE - think of a better name than vrec!
+   */
+     
+  EXPORT VRec(virtual record outrec) := MODULE
+
+    SHARED externals := SERVICE : fold
+      STRING dumpRecordType(virtual record val) : eclrtl,pure,library='eclrtl',entrypoint='dumpRecordType';
+      DATA serializeRecordType(virtual record val) : eclrtl,pure,library='eclrtl',entrypoint='serializeRecordType';
+
+      streamed dataset(outrec) translate(streamed dataset input) : eclrtl,pure,library='eclrtl',entrypoint='transformDataset',passParameterMeta(true);
+      _linkcounted_ row(outrec) translateRow(virtual record outrec, ROW indata) : eclrtl,pure,library='eclrtl',entrypoint='transformRecord',passParameterMeta(true);
+
+      DATA serializeRow(virtual record outrec, ROW inRow) : eclrtl,pure,library='eclrtl',entrypoint='serializeRow',passParameterMeta(true),fold;
+      _linkcounted_ row(outrec) deserializeRow(virtual record outrec, DATA parcel) : eclrtl,pure,library='eclrtl',entrypoint='deserializeRow',passParameterMeta(true),fold;
+      
+      SET OF STRING getFieldNames(virtual record meta) : eclrtl,pure,library='eclrtl',entrypoint='getFieldNames',fold,passParameterMeta(true);
+      UNSIGNED4 getNumFields(virtual record meta) : eclrtl,pure,library='eclrtl',entrypoint='getNumFields',fold,passParameterMeta(true);
+
+      unsigned4 getFieldNum(virtual record meta, CONST VARSTRING name) : eclrtl,pure,library='eclrtl',entrypoint='getFieldNum',fold,passParameterMeta(true);
+      string getFieldName(virtual record meta, unsigned4 fieldNum) : eclrtl,pure,library='eclrtl',entrypoint='getFieldName',fold,passParameterMeta(true);
+      string getFieldType(virtual record meta, unsigned4 fieldNum) : eclrtl,pure,library='eclrtl',entrypoint='getFieldType',fold,passParameterMeta(true);
```

Comment:
A human readable string for field type is harder to use programmatically.  Consider an ENUM or at least a binary value that is easier to use in CASE or MAP or IF functions.  There should be a function to produce a readable name for debugging outputs.